File: project/sungwoong-onboarding/src/main/java/com/multi/sungwoongonboarding/questions/infrastructure/QuestionJpaEntity.java
package com.multi.sungwoongonboarding.questions.infrastructure;

import com.multi.sungwoongonboarding.common.entity.BaseEntity;
import com.multi.sungwoongonboarding.forms.infrastructure.FormsJpaEntity;
import com.multi.sungwoongonboarding.options.infrastructure.OptionsJpaEntity;
import com.multi.sungwoongonboarding.questions.domain.Questions;
import jakarta.persistence.*;
import lombok.Getter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Entity
@Table(name = "questions")
public class QuestionJpaEntity extends BaseEntity {

    @Id
    @Column(name = "question_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "question_text", nullable = false)
    private String questionText;

    @Column(name = "question_type", nullable = false)
    private String questionType;

    @Column(name = "question_order", nullable = false)
    private int order;

    @Column(name = "is_required", nullable = false)
    private boolean isRequired;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "form_id")
    private FormsJpaEntity formsJpaEntity;

    @OneToMany(mappedBy = "questionJpaEntity", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OptionsJpaEntity> options = new ArrayList<>();


    public static QuestionJpaEntity fromDomain(Questions questions) {

        QuestionJpaEntity questionJpaEntity = new QuestionJpaEntity();
        questionJpaEntity.id = questions.getId();
        questionJpaEntity.questionText = questions.getQuestionText();
        questionJpaEntity.questionType = questions.getQuestionType().name();
        questionJpaEntity.order = questions.getOrder();
        questionJpaEntity.isRequired = questions.isRequired();
        return questionJpaEntity;
    }

    public static Questions toDomain(QuestionJpaEntity questionJpaEntity) {
        return Questions.builder()
                .id(questionJpaEntity.getId())
                .questionText(questionJpaEntity.getQuestionText())
                .questionType(Questions.QuestionType.valueOf(questionJpaEntity.getQuestionType()))
                .order(questionJpaEntity.getOrder())
                .isRequired(questionJpaEntity.isRequired())
                .build();
    }

    public void mappingFormJpaEntity(FormsJpaEntity formsJpaEntity) {
        this.formsJpaEntity = formsJpaEntity;
        if (!formsJpaEntity.getQuestions().contains(this)) {
            formsJpaEntity.getQuestions().add(this);
        }
    }
}

File: project/taehyung-onboarding/src/main/kotlin/formService/adapter/port/outbound/persistence/jpa/QuestionJpaEntity.kt
package formService.adapter.port.outbound.persistence.jpa

import formService.domain.Question
import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToMany
import jakarta.persistence.Table

@Entity
@Table(name = "question")
class QuestionJpaEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    @Column
    var name: String,
    @Column
    var description: String,
    @Enumerated(value = EnumType.STRING)
    var inputType: Question.QuestionInputType,
    @Column
    var required: Boolean,
) {
    @OneToMany(mappedBy = "question", cascade = [CascadeType.PERSIST])
    protected val mutableOptions: MutableList<QuestionOptionJpaEntity> = mutableListOf()
    val options: List<QuestionOptionJpaEntity> get() = mutableOptions.toList()

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_form_id")
    var surveyForm: SurveyFormJpaEntity? = null

    fun addOptions(options: List<QuestionOptionJpaEntity>?) {
        options?.forEach {
            mutableOptions.add(it)
            it.question = this
        }
    }

    override fun toString(): String =
        "QuestionJpaEntity(id=$id, name='$name', description='$description', inputType=$inputType, required=$required, mutableOptions=$mutableOptions, options=$options)"
}
